### PR TITLE
recording

### PR DESCRIPTION
**Pull request by alichraghi**
- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

File: src/main.zig [hunk 1]
```diff
@@ -245,10 +258,11 @@ pub const Player = struct {
...
 
     pub fn sampleRate(self: Player) u24 {
-        return if (@hasField(Backend, "jack")) switch (self.data) {
-            .jack => |b| b.sampleRate(),
-            inline else => |b| b.sample_rate,
-        } else switch (self.data) {
+        // return if (@hasField(Backend, "jack")) switch (self.data) {
+        //     .jack => |b| b.sampleRate(),
+        //     inline else => |b| b.sample_rate,
+        // } else
```
Comment (emidoots) [inline src/main.zig]: Why this comment?

**Review by emidoots — APPROVED** [review]

File: src/main.zig [hunk 2]
```diff
@@ -272,6 +286,167 @@ pub const Player = struct {
...
+            inline else => |b| b.paused(),
+        };
+    }
+
+    pub const SetVolumeError = error{
+        CannotSetVolume,
+    };
+
+    // confidence interval (±) depends on the device
+    pub fn setVolume(self: Recorder, vol: f32) SetVolumeError!void {
```
Comment (emidoots) [inline src/main.zig]: I am curious/doubtful that other APIs will support pausing, setting volume, etc. when recording. Will be interesting to see

Comment (alichraghi) [reply src/main.zig]: yeah im not sure too. but adding them wasn't a big deal anyway.

File: src/main.zig (hunk 1 above)
Comment (alichraghi) [reply src/main.zig]: compile error. (temporary)

**Comment by alichraghi** [comment]
im trying to implement this on as many backends as i can

**Pull request closed without merging**